spynnaker/pyNN/utilities/utility_calls.py

```diff
@@ -40,7 +40,11 @@ def convert_param_to_numpy(param, no_atoms):
 
         # numpy reduces a single valued array to a single value, so enforce
         # that it is an array
-        return numpy.array([param.next(n=no_atoms)], dtype="float")
+        param_value = param.next(n=no_atoms)
+        if hasattr(param_value, '__iter__'):
+            return numpy.array(param_value, dtype="float")
+        else:
+            return numpy.array([param.next(n=no_atoms)], dtype="float")
 
```

Comment:
i ack that the param.next() in the else needs to disappear. that said, pynn 0.7 still needs to be detected, so you'd need something like this.